src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts

```diff
@@ -398,6 +401,14 @@ export class CommandDetectionCapability extends Disposable implements ICommandDe
 		}
 		this._currentCommand = new PartialTerminalCommand(this._terminal);
 		this._handleCommandStartOptions = undefined;
+
+		if (this._lspTerminalDictionaryService) {
+			const documentModel: MutableDisposable<ILspTerminalModelContentProvider> | undefined = this._lspTerminalDictionaryService.get('randomTerminalId');
+			if (documentModel) {
+				// TODO: Have another mapping for terminalID to Virtual Document URI
+				documentModel.value?.setContent(createTerminalLanguageVirtualUri('1', 'py'), newCommand?.command!);
+			}
+		}
```

Comment:
this can be moved into the lsp provider addon, where you have access to the command detection capability (will need to pass that in)